File: app/src/main/java/kr/djspi/pipe01/tab/PreviewTab.java
package kr.djspi.pipe01.tab;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.gson.JsonObject;
import com.helloelliote.util.json.Json;

import kr.djspi.pipe01.R;

import static kr.djspi.pipe01.Const.RESULT_FAIL;
import static kr.djspi.pipe01.Const.RESULT_PASS;
import static kr.djspi.pipe01.Const.TAG_PREVIEW;

public class PreviewTab extends Fragment implements OnClickListener {

    private static final String TAG = PreviewTab.class.getSimpleName();
    private JsonObject jsonObject;
    private Uri imageFileUri;
    private OnRecordListener listener;

    public PreviewTab() {
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnRecordListener) {
            listener = (OnRecordListener) context;
            jsonObject = listener.getJsonObject();
            imageFileUri = listener.getUri();
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.tab_preview, container, false);

        TextView tPipe = view.findViewById(R.id.text_pipe);
        tPipe.setText(Json.s(jsonObject, "pipe"));

        TextView tShape = view.findViewById(R.id.text_shape);
        tShape.setText(Json.s(jsonObject, "shape"));

        TextView tHorizontal = view.findViewById(R.id.text_horizontal);
        tHorizontal.setText(Json.s(jsonObject, "horizontal_form"));

        TextView tVertical = view.findViewById(R.id.text_vertical);
        tVertical.setText(Json.s(jsonObject, "vertical_form"));

        TextView tDepth = view.findViewById(R.id.text_depth);
        tDepth.setText(Json.s(jsonObject, "depth"));

        TextView tHeader = view.findViewById(R.id.header);
        tHeader.setText(Json.s(jsonObject, "header"));

        TextView tSpec = view.findViewById(R.id.text_spec);
        tSpec.setText(Json.s(jsonObject, "spec"));

        TextView tUnit = view.findViewById(R.id.unit);
        tUnit.setText(Json.s(jsonObject, "unit"));

        TextView tMaterial = view.findViewById(R.id.text_material);
        tMaterial.setText(Json.s(jsonObject, "material"));

        TextView tSupervise = view.findViewById(R.id.text_supervise);
        tSupervise.setText(Json.s(jsonObject, "supervise"));

        TextView tSuperviseContact = view.findViewById(R.id.text_supervise_contact);
        tSuperviseContact.setText(Json.s(jsonObject, "supervise_contact"));

        TextView tMemo = view.findViewById(R.id.text_memo);
        tMemo.setText(Json.s(jsonObject, "spi_memo"));

        TextView tConstruction = view.findViewById(R.id.text_construction);
        tConstruction.setText(Json.s(jsonObject, "construction"));

        TextView tConstructionContact = view.findViewById(R.id.text_construction_contact);
        tConstructionContact.setText(Json.s(jsonObject, "construction_contact"));

        TextView tPhoto = view.findViewById(R.id.text_photo);
        if (imageFileUri != null) {
            tPhoto.setText(getString(R.string.record_photo_ok));
        } else tPhoto.setText(null);

        view.findViewById(R.id.button_confirm).setOnClickListener(this);

        return view;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_confirm:
                listener.onRecord(TAG_PREVIEW, RESULT_PASS);
                break;
            default:
                listener.onRecord(TAG_PREVIEW, RESULT_FAIL);
                break;
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }
}

File: app/src/main/java/kr/djspi/pipe01/tab/PreviewTab.kt
package kr.djspi.pipe01.tab

import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.gson.JsonObject
import kr.djspi.pipe01.Const.RESULT_FAIL
import kr.djspi.pipe01.Const.RESULT_PASS
import kr.djspi.pipe01.Const.TAG_PREVIEW
import kr.djspi.pipe01.R
import kr.djspi.pipe01.databinding.TabPreviewBinding

class PreviewTab : Fragment(), View.OnClickListener {

    private var _binding: TabPreviewBinding? = null
    private val binding get() = _binding!!
    private lateinit var listener: OnRecordListener
    private lateinit var json: JsonObject
    private var imageFileUri: Uri? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnRecordListener) {
            listener = context
            json = listener.jsonObject
            imageFileUri = listener.uri
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = TabPreviewBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.textPipe.text = json["pipe"].asString
        binding.textShape.text = json["shape"].asString
        binding.textHorizontal.text = json["horizontal_form"].asString
        binding.textVertical.text = json["vertical_form"].asString
        binding.textDepth.text = json["depth"].asString
        binding.header.text = json["header"].asString
        binding.textSpec.text = json["spec"].asString.replace("^", " ")
        binding.unit.text = json["unit"].asString
        binding.textMaterial.text = json["material"].asString.replace("^", " ")
        binding.textSupervise.text = json["supervise"].asString
        binding.textSuperviseContact.text = json["supervise_contact"].asString
        binding.textMemo.text = json["spi_memo"].asString
        binding.textConstruction.text = json["construction"].asString
        binding.textConstructionContact.text = json["construction_contact"].asString

        if (imageFileUri != null) {
            binding.textPhoto.text = getString(R.string.record_photo_ok)
        } else {
            binding.textPhoto.text = null
        }

        binding.buttonNext.setOnClickListener(this)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.button_next -> listener.onRecord(TAG_PREVIEW, RESULT_PASS)
            else -> listener.onRecord(TAG_PREVIEW, RESULT_FAIL)
        }
    }
}
